### ISSUE TITLE
LineLengthCheck check doubles non-ASCII symbols

### ISSUE THREAD

**Issue by proshin-roman** [teamed/qulice#873]
**What's wrong**
I have next line in the code (keeping indentation):
```!java
        final String input = "Hello, товарищ output äÄ üÜ öÖ and ß";
```
This line has 68 chars length and contains 14 non-ASCII symbols.
Then I run the build with qulice maven plugin (version 0.17.1; OS Windows 10; standard command line) and get next error:
```
[INFO] Checkstyle: <class name>.java[83]: Line is longer than 80 characters (found 82). (LineLengthCheck)
```
It says the length is 82 that means that non-ASCII symbols are calculated twice: 68 (including non-ASCII) + 14 (doubled non-ASCII symbols) = 82. It's a bug!

**How it should be**
 All symbols must be calculated in same way and once only.

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) please, pay attention to this issue

**Comment by krzyk** [comment]
@proshin-roman The issue here is actually in checkstyle, have you checked there if they already fixed this bug?

**Comment by proshin-roman** [comment]
@krzyk It might be this issue https://github.com/checkstyle/checkstyle/issues/5089

**Comment by krzyk** [comment]
@proshin-roman So not much we can do here right now until it is fixed in checkstyle :(

**Comment by proshin-roman** [comment]
@krzyk Yes, I agree but I would prefer to keep the ticket open so that we don't forget about this problem. And then it can be closed as soon as qulice is upgraded to the fixed Checkstyle version.

**Comment by lrozenblyum** [comment]
The bug is reproduced in the build: http://www.rultor.com/t/14467-387496260

**Comment by QwerTech** [comment]
+1 for this issue.

**Comment by yegor256** [comment]
Checked on `master`: this bug is no longer reproducible with the current Checkstyle version used by Qulice (12.3.1). `LineLengthCheck` now counts characters, not UTF-8 bytes, so non-ASCII symbols are counted once.

Added a regression test in 2261830c4 (`CheckstyleLineLengthTest#doesNotDoubleCountNonAsciiChars`) with a resource file containing a line that is 99 characters but 144 UTF-8 bytes. If the upstream behaviour ever regresses, the test will fail.

Closing as fixed.
